plugins/modules/inventory_intent.py

```diff
@@ -1992,13 +2033,114 @@ def get_device_ids(self, device_ips):
             except Exception as e:
                 error_message = "Error while fetching device '{0}' from Cisco Catalyst Center: {1}".format(device_ip, str(e))
                 self.log(error_message, "ERROR")
-                raise Exception(error_message)
 
         return device_ids
 
+    def get_device_ips_from_hostname(self, hostname_list):
+        """
+        Get the list of unique device IDs for list of specified hostnames of devices in Cisco Catalyst Center.
+        Parameters:
+            self (object): An instance of a class used for interacting with Cisco Catalyst Center.
+            hostname_list (list): The hostnames of devices for which you want to retrieve the device IDs.
+        Returns:
+            list: The list of unique device IDs for the specified devices hostname list.
+        Description:
+            Queries Cisco Catalyst Center to retrieve the unique device ID associated with a device having the specified
+            list of hostnames. If the device is not found in Cisco Catalyst Center, tthen print the log message with error severity.
```

Comment:
Spelling "tthen" 
Or we can simply write as 
If a device is not found in Cisco Catalyst Center, an error log message is printed.

Reply:
updated.